pkg/xds/generator/ingress_generator.go

```diff
@@ -112,11 +118,33 @@ func (i IngressGenerator) generateLDS(
 	return inboundListenerBuilder.Build()
 }
 
+func tagsFromTargetRef(targetRef common_api.TargetRef) (tags.Tags, bool) {
+	var service string
+	var tags tags.Tags
+
+	switch targetRef.Kind {
+	case common_api.MeshService:
+		service = targetRef.Name
+	case common_api.MeshServiceSubset:
+		service = targetRef.Name
+		tags = targetRef.Tags
+	case common_api.Mesh:
+		service = mesh_proto.MatchAllTag
+	case common_api.MeshSubset:
+		service = mesh_proto.MatchAllTag
+		tags = targetRef.Tags
+	default:
+		return nil, false
+	}
+
+	return tags.WithTags(mesh_proto.ServiceTag, service), true
+}
+
 func (_ IngressGenerator) destinations(
 	ingressProxy *core_xds.ZoneIngressProxy,
 ) map[string][]tags.Tags {
 	destinations := map[string][]tags.Tags{}
-	for _, tr := range ingressProxy.TrafficRouteList.Items {
+	for _, tr := range ingressProxy.PolicyResources[core_mesh.TrafficRouteType].(*core_mesh.TrafficRouteResourceList).Items {
```

Comment:
I think it's going to panic if there are no entries for `core_mesh.TrafficRouteType`. Probably it's better to check `len()` as you do for `meshhttproute_api.MeshHTTPRouteType`

Reply:
I don't check `len` before `range`ing through the list though. I set the entry for `TrafficRouteType` unconditionally in the map in `pkg/xds/sync/ingress_proxy_builder.go`

Reply:
That is to say, it will only panic if `ingressProxy.PolicyResources[core_mesh.TrafficRouteType]` is `nil` (or the wrong type of course). But this is the same as before, where `ingressProxy.TrafficRouteList` is assumed to not be `nil`.
 `Items` being `nil` or empty isn't a problem, is it?